examples/capture_video_nframes.py

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/python3
+import time
+from threading import Condition
+
+from picamera2 import Picamera2
+from picamera2.encoders import H264Encoder
+
+picam2 = Picamera2()
+video_config = picam2.create_video_configuration()
+picam2.configure(video_config)
+
+
+class H264EncoderStop(H264Encoder):
+
+    def __init__(self, bitrate=None, repeat=True, iperiod=None, framerate=None, enable_sps_framerate=False, qp=None):
+        super().__init__(bitrate, repeat, iperiod, framerate, enable_sps_framerate, qp)
+        self.frame_counter = 0
+        self.cond = Condition()
+
+    def _encode(self, stream, request):
+        self.frame_counter += 1
+        # Stop capturing when 200 frames is reached
+        if self.frame_counter <= 200:
+            super()._encode(stream, request)
+            if self.frame_counter == 200:
+                with self.cond:
+                    self.cond.notify()
+
+
+encoder = H264EncoderStop(10000000)
+picam2.start_recording(encoder, 'test.h264')
+
+# Wait for 200 frames to be captured
+with encoder.cond:
+    encoder.cond.wait()
+
+# Necessary to ensure all frames are written
+time.sleep(1)
```

Comment:
Just wondering if `picam2.stop_recording()` would ensure everything gets flushed? Then we could perhaps dispense with the sleep, which feels slightly awkward...

Reply:
Strangely it doesn't, I just tried that now and got 199 frames written, I agree the sleep is awkward

Reply:
That's unfortunate.  What about doing the counting in the encoder's `outputframe` method? By the end of that function the output has actually been given the frame, so hopefully it will get flushed before it stops?

Reply:
Thanks, that did the trick :) I've moved to subclassing the FileOutput class instead and stopping that.

Reply:
Cool, I'll let the tests run through and then merge it.